src/hooks/useCustomNotification.ts

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react';
+
+import { uniqueId } from 'lodash';
+
+import {
+  DEFAULT_TOAST_AUTO_CLOSE_MS,
+  NotificationDisplayData,
+  NotificationType,
+} from '@/constants/notifications';
+
+import { useAppDispatch } from '@/state/appTypes';
+import { addCustomNotification } from '@/state/notifications';
+
+export const useCustomNotification = () => {
+  const dispatch = useAppDispatch();
+
+  const notify = useCallback(
+    (customNotification: Omit<NotificationDisplayData, 'groupKey'>) => {
+      dispatch(
+        addCustomNotification({
+          id: uniqueId(),
```

Comment:
open to other ID generators here too, anything random is fine

Reply:
I have used `crypto?.randomUUID() ?? Date.now().toString()` elsewhere if we don't want to depend on lodash, but don't feel strongly

Reply:
oh i like Date.now() :D